cfep-XX.md

```diff
@@ -0,0 +1,61 @@
+
+<table>
+<tr><td> Title </td><td> Guidelines for renaming packages </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td>
+    Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt; Matthew R. Becker &lt;becker.mr@gmail.com&gt
+</td></tr>
+<tr><td> Created </td><td> Oct 26, 2025 </td></tr>
+<tr><td> Updated </td><td> Oct 26, 2025 </td></tr>
+<tr><td> Discussion </td><td> https://github.com/conda-forge/admin-requests/pull/1711 </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+This CFEP describes the principles informing how and when to rename a package in conda-forge.
+Two cases are described depending on the status of the target name:
+
+- Unclaimed names: A package already present in conda-forge wants to use a new name currently unclaimed.
+- Names in use: A package, already present or about to be submitted, wishes to use a name already in use.
+
+## Specification
+
+This CFEP follows [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) in its use of certain keywords (e.g., MUST).
+
+### Feedstock Epochs
+
+In order to track how feedstocks are reused, we define the concept of the `feedstock epoch` and a `feedstock epoch label`.
+
+- When a feedstock is first created, its epoch is 0.
+- If a feedstock is reused for a new project (see below), the `feedstock epoch` MUST be incremented by 1.
+- A `feedstock epoch label` is a label of the form `{feedstock name without '-feedstock'}_feedstock_epoch{N}`. When the label is applied to an artifact, the label MUST reflect epoch of the feedstock at the time it produced the given artifact.
+- All artifacts uploaded from a feedstock which do not otherwise have a `feedstock epoch label` are assumed to be from the current `feedstock epoch`.
+- A `feedstock epoch label` MUST be applied to any artifact from the previous epoch of the feedstock.
+- The current `feedstock epoch` for a given feedstock is determined as follows. The set of all labels for all artifacts produced by the feedstock is searched for any label matching the form of a `feedstock epoch label` for that feedstock. If no label matches, then the current `feedstock epoch` is 0. If labels do match, then the current `feedstock epoch` is one plus the maximum `feedstock epoch` over all matching labels.
+
+### Unclaimed names
+
+When an existing package wants to be renamed to a currently unclaimed name, the process is uncontroversial and straight-forward. Two approaches are possible:
+
+- The new name needs to be registered to the originating feedstock in `feedstock-outputs` (or equivalent). Once approved, a PR in the feedstock adding the new output may proceed normally. The old name MUST be kept as an alias output for backwards compatibility. In this case, the feedstock retains the name of the original name.
```

Comment:
I think we should not mandate keeping these aliases indefinitely. In fact, it's often an explicit goal to move away from these. Of course, the compat output is necessary to help with the transition, but for various renames (e.g. `boost-cpp` -> `libboost[-devel]`, `arrow-cpp` -> `libarrow`, etc.) the compat outputs were dropped after a year or so, and I consider this right and proper.

One way that we've been cleaning up after ourselves is to add a linter rule that warns upon usage of the old names.